app/models/start_teacher_training.rb

```diff
@@ -1,13 +1,18 @@
 class StartTeacherTraining < Base
   attribute :year_of_entry, :string
 
+  validates :year_of_entry, types: { method: :get_candidate_initial_teacher_training_years, message: "You must select an option from the list" }
   validate :date_cannot_be_in_the_past, unless: :dont_know
 
-  def year_range(number_of_years) # sets year range for view
-    years = (Date.today.year..Date.today.next_year(number_of_years).year).map do |year|
-      OpenStruct.new(value: year, name: year)
-    end
-    years << OpenStruct.new(value: "don't know", name: "Don't know")
+  def self.options
+    option_list = {}
+    ApiClient::get_candidate_initial_teacher_training_years.each { |type| option_list[type.value] = type.id }
+    option_list
+  end
+
+  def year_range(number_of_years) # sets year range for view, this must be within api range!
+    years = ApiClient.get_candidate_initial_teacher_training_years
+    years.select { |year| year.value == "Not sure" || year.value.to_i.between?(Date.today.year, Date.today.next_year(number_of_years).year) }
```

Comment:
Should probably use the id of the type to determine if it's the `Not sure` option here

Reply:
ok